File: src/libs/ThirdParty.VBSuppliers/Properties/Travelgate/ITravelgateSettings.vb
Imports ThirdParty.Abstractions

Public Interface ITravelgateSettings
    ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property URL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property SearchSOAPAction(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property PrebookSOAPAction(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property BookSOAPAction(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property CancelSOAPAction(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property UseGZip(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer
    ReadOnly Property RequiresVCard(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property ReferenceDelimiter(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As String
    ReadOnly Property DefaultNationality(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As String
    ReadOnly Property CardHolderName(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property EncryptedCardDetails(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property ProviderUsername(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property ProviderPassword(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property ProviderCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property UrlReservation(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property UrlGeneric(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property UrlValuation(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property UrlAvail(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property LanguageCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Parameters(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Markets(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property CurrencyCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property MaximumHotelSearchNumber(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property MaximumCitySearchNumber(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property MaximumRoomNumber(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property MaximumRoomGuestNumber(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property MinimumStay(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property AllowHotelSearch(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property UseZoneSearch(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property SearchRequestTimeout(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property RatePlanCodes(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property SendGUIDReference(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
End Interface

File: src/libs/iVectorOne.Suppliers/Travelgate/ITravelgateSettings.cs
namespace iVectorOne.Suppliers.Travelgate
{
    public interface ITravelgateSettings
    {
        string User(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string Password(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string GenericURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SearchURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string PrebookURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string BookingURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string CancellationURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        bool UseGZip(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        bool RequiresVCard(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string ReferenceDelimiter(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string LeadGuestNationality(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string CardHolderName(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string EncryptedCardDetails(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SecondaryUsername(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SecondaryPassword(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string ProviderCode(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SecondaryBookURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SecondaryGenericURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SecondaryPrebookURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SecondarySearchURL(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string LanguageCode(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string Parameters(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string SourceMarket(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string Currency(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int HotelBatchLimit(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int CityBatchLimit(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int RoomSearchLimit(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int MaximumRoomGuestNumber(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int MinimumStay(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        bool EnableHotelSearch(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        bool UseZoneSearch(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        int SearchRequestTimeout(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        string RatePlanCodes(IThirdPartyAttributeSearch tpAttributeSearch, string source);
        bool SendGUIDReference(IThirdPartyAttributeSearch tpAttributeSearch, string source);
    }
}